src/plots/cartesian/tick_label_defaults.js

```diff
@@ -50,6 +50,8 @@ module.exports = function handleTickLabelDefaults(containerIn, containerOut, coe
     }
 
     if(axType !== 'category' && !options.noHover) coerce('hoverformat');
+
+    if(axType === 'cartesian') coerce('ticklabelsautomargin');
```

Comment:
Oh does `coerce('ticklabelsautomargin')` have an effect when `showticklabels` is false? More precisely, do the axis title font settings and/or `ticklen` affect to auto-margin computations?

Reply:
I'm not sure what `coerce` actually does, to be honest, so I'm not sure if it will have an impact. What exactly does it do? Is it necessary here?

To answer your question more generally, the auto-margin calculation explicitly takes the axis title font size into account, and for the rest relies on the pre-existing bounding-box calculation.

Reply:
> I'm not sure what coerce actually does,

In brief, `coerce` fills in `fullLayout` with a sanitise version of its user layout value. Almost everywhere downstream of the `Plots.supplyDefaults` uses `fullLayout` and `fullData` value instead of `gd.layout` and `gd.data` including your new logic in `axes.js`. 

So, yes it is necessary :smile: 

>  the auto-margin calculation explicitly takes the axis title font size into account,

Ok great. So, auto-margin doesn't _just_ depend on tick settings like the attribute name `ticklabelsautomargin` is suggesting. We'll need to think about how to name this new attribute some more cc https://github.com/plotly/plotly.js/pull/2243#discussion_r165100920

> for the rest relies on the pre-existing bounding-box calculation.

Ok, so this should depend on `ticklen` (we should double-check though). Knowing which attributes  impact the auto-margin computation is important to make sure relayout calls like `Plotly.relayout(gd, 'xaxis.ticklen', /**/)` update the (auto) margin when `ticklabelsautomargin` is turned on.